src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -618,6 +618,37 @@ impl LogicalAgg {
             .iter()
             .any(|call| matches!(call.agg_kind, AggKind::StringAgg | AggKind::ArrayAgg))
     }
+
+    fn output_requires_order(&self, required_order: &Order) -> bool {
```

Comment:
please add some comments here. also on `input_supplies_order`

Reply:
I also renamed the functions, I hope it is more clear now